scripts/upgrade-via-npm.sh

```
@@ -1,652 +1,845 @@
+    run_with_timeout "$INSTALL_TIMEOUT" "plugins install (local dir)" \
+        openclaw plugins install "$package_dir" $FORCE_UNSAFE_FLAG 2>&1 || rc=$?
+
+    rm -rf "$extract_dir" 2>/dev/null || true
+
+    if [ $rc -eq 0 ] && [ -f "$EXTENSIONS_DIR/$PLUGIN_ID/package.json" ]; then
+        echo "  ✅ Level 2 安装成功"
+        return 0
+    fi
+    echo "  Level 2 失败 (exit=$rc)"
+    [ -d "$EXTENSIONS_DIR/$PLUGIN_ID" ] && [ ! -f "$EXTENSIONS_DIR/$PLUGIN_ID/package.json" ] && \
+        rm -rf "$EXTENSIONS_DIR/$PLUGIN_ID" 2>/dev/null || true
```

Shellcheck (0.11.0)

[warning] Use "${var:?}" to ensure this never expands to / .

(SC2115)